### PR TITLE
sql: fix sequence doc

### PR DESCRIPTION
**Pull request by AilinKid**
### What is changed, added or deleted? (Required)
Add a description of how does TiDB setval differ from Maraidb.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):https://github.com/pingcap/docs-cn/pull/2282  https://github.com/pingcap/tidb/pull/14829

### CODE REVIEW

File: reference/sql/statements/create-sequence.md [hunk 1]
```diff
@@ -272,6 +272,16 @@ ERROR 4135 (HY000): Sequence 'test.seq2' has run out
 
 * MySQL 暂无 SEQUENCE 选项。TiDB Sequence 借鉴自 MariaDB，但是 setval 会保持原有的步调。
 
+这里的步调是指，sequence 序列中的数在定义之后会产生一定的等差关系。setval 虽然可以将 sequence 的当前值进行移动设置，但是后续出现的值仍会遵循原有的等差关系。
+
```
Comment (TomShawn) [inline reference/sql/statements/create-sequence.md]:
```suggestion

示例如下：

```

File: reference/sql/statements/create-sequence.md [hunk 2]
```diff
@@ -272,6 +272,16 @@ ERROR 4135 (HY000): Sequence 'test.seq2' has run out
...
+```
+1, 3, 5, ...            // 序列遵循起始为 1，步长为 2 的等差关系
```
Comment (TomShawn) [inline reference/sql/statements/create-sequence.md]:
```suggestion
1, 3, 5, ...            // 序列遵循起始为 1、步长为 2 的等差关系。
```

File: reference/sql/statements/create-sequence.md [hunk 3]
```diff
@@ -272,6 +272,16 @@ ERROR 4135 (HY000): Sequence 'test.seq2' has run out
...
+select setval(seq, 6)   // 设置 sequence 的当前值为 6
```
Comment (TomShawn) [inline reference/sql/statements/create-sequence.md]:
```suggestion
select setval(seq, 6)   // 设置 sequence 的当前值为 6。
```

File: reference/sql/statements/create-sequence.md [hunk 4]
```diff
@@ -272,6 +272,16 @@ ERROR 4135 (HY000): Sequence 'test.seq2' has run out
 
 * MySQL 暂无 SEQUENCE 选项。TiDB Sequence 借鉴自 MariaDB，但是 setval 会保持原有的步调。
 
+这里的步调是指，sequence 序列中的数在定义之后会产生一定的等差关系。setval 虽然可以将 sequence 的当前值进行移动设置，但是后续出现的值仍会遵循原有的等差关系。
```
Comment (TomShawn) [inline reference/sql/statements/create-sequence.md]:
```suggestion
  这里的步调是指，SEQUENCE 序列中的数在定义之后会产生一定的等差关系。setval 虽然可以将 SEQUENCE 的当前值进行移动设置，但是后续出现的值仍会遵循原有的等差关系。
```

File: reference/sql/statements/create-sequence.md [hunk 5]
```diff
@@ -272,6 +272,16 @@ ERROR 4135 (HY000): Sequence 'test.seq2' has run out
...
+
+```
+1, 3, 5, ...            // 序列遵循起始为 1，步长为 2 的等差关系
+select setval(seq, 6)   // 设置 sequence 的当前值为 6
+7, 9, 11, ...           // 后续产生值仍会遵循这个等差关系
```
Comment (TomShawn) [inline reference/sql/statements/create-sequence.md]:
```suggestion
7, 9, 11, ...           // 后续产生值仍会遵循这个等差关系。
```

File: reference/sql/statements/create-sequence.md [hunk 6]
```diff
@@ -272,6 +272,16 @@ ERROR 4135 (HY000): Sequence 'test.seq2' has run out
...
+```
+
+对于 sequence 的起始值，在 cycle 模式下，第一轮会是 start，后续轮次将会是 MinValue (increment > 0) / MaxValue (increment < 0)。
```
Comment (TomShawn) [inline reference/sql/statements/create-sequence.md]:
```suggestion
在 cycle 模式下，SEQUENCE 的起始值第一轮为 `start`，后续轮次将会是 `MinValue (increment > 0)`/` MaxValue (increment < 0)`。
```

**Comment by TomShawn** [comment]
@AilinKid [#2282](https://github.com/pingcap/docs-cn/pull/2282) 中新增的三篇 `SEQUENCE` 文档需要添加到 TOC 吗？

**Comment by AilinKid** [comment]
> 文档需要添加到 TOC 吗？

需要的，谢谢哈

**Comment by TomShawn** [comment]
LGTM
@AilinKid Please involve a technical review, thanks!

File: TOC.md [hunk 7]
```diff
@@ -151,6 +151,7 @@
       - [`CREATE TABLE`](/reference/sql/statements/create-table.md)
       - [`CREATE USER`](/reference/sql/statements/create-user.md)
       - [`CREATE VIEW`](/reference/sql/statements/create-view.md)
+      - [`CREATE SEQUENCE`](/reference/sql/statements/create-sequence.md)
```
Comment (djshow832) [inline TOC.md]: It's not in order. It should be placed after `CREATE INDEX`.

File: TOC.md [hunk 8]
```diff
@@ -162,6 +163,7 @@
       - [`DROP TABLE`](/reference/sql/statements/drop-table.md)
       - [`DROP USER`](/reference/sql/statements/drop-user.md)
       - [`DROP VIEW`](/reference/sql/statements/drop-view.md)
+      - [`DROP SEQUENCE`](/reference/sql/statements/drop-sequence.md)
```
Comment (djshow832) [inline TOC.md]: Ditto.

File: TOC.md [hunk 9]
```diff
@@ -189,6 +191,7 @@
       - [`SHOW COLLATION`](/reference/sql/statements/show-collation.md)
       - [`SHOW [FULL] COLUMNS FROM`](/reference/sql/statements/show-columns-from.md)
       - [`SHOW CREATE TABLE`](/reference/sql/statements/show-create-table.md)
+      - [`SHOW CREATE SEQUENCE`](/reference/sql/statements/show-create-sequence.md)
```
Comment (djshow832) [inline TOC.md]: Ditto.

File: reference/sql/statements/create-sequence.md [hunk 10]
```diff
@@ -54,32 +56,34 @@ CREATE [TEMPORARY] SEQUENCE [IF NOT EXISTS] sequence_name
...
+## `SEQUENCE` 函数
 
-SEQUENCE 的使用主要通过表达式函数来操纵
+主要通过表达式函数来操纵 SEQUENCE 的使用。
 
-**NEXTVAL / NEXT VALUE FOR**
++ `NEXTVAL` 或 `NEXT VALUE FOR`
 
-本质上都是 nextval() 函数, 获取 SEQUENCE 对象的下一个有效值，其参数为 SEQUENCE 序列的 identifier。
+    本质上都是 `nextval(`) 函数，获取 SEQUENCE 对象的下一个有效值，其参数为 `SEQUENCE` 序列的 `identifier`。
```
Comment (djshow832) [inline reference/sql/statements/create-sequence.md]:
```suggestion
    本质上都是 `nextval()` 函数，获取 `SEQUENCE` 对象的下一个有效值，其参数为 `SEQUENCE` 序列的 `identifier`。
```

File: reference/sql/statements/create-sequence.md [hunk 11]
```diff
@@ -270,7 +274,19 @@ ERROR 4135 (HY000): Sequence 'test.seq2' has run out
...
+
+这里的步调是指，SEQUENCE 序列中的数在定义之后会产生一定的等差关系。`SETVAL` 虽然可以将 SEQUENCE 的当前值进行移动设置，但是后续出现的值仍会遵循原有的等差关系。
+
+示例如下：
+
+```
+1, 3, 5, ...            // 序列遵循起始为 1、步长为 2 的等差关系。
+select setval(seq, 6)   // 设置 SEQUENCE 的当前值为 6。
+7, 9, 11, ...           // 后续产生值仍会遵循这个等差关系。
+```
```
Comment (djshow832) [inline reference/sql/statements/create-sequence.md]: Please use some SQLs to indicate this.
e.g. select nextval...(show output) ; select setval ....; select nextval ...(show output).

Comment (AilinKid) [reply reference/sql/statements/create-sequence.md]: here is ok, this wants to explain the logic of the numbers, using the SQL show here will be a huge page.

**Comment by AilinKid** [comment]
Thank you so much @TomShawn for this.
PTAL @djshow832

**Pull request merged**